packages/flutter_tools/lib/src/project.dart

```diff
@@ -509,12 +509,10 @@ class AndroidProject extends FlutterProjectPlatform {
     if (plugin.existsSync()) {
       return false;
     }
-    final File appGradle = hostAppGradleRoot.childFile(
-        fileSystem.path.join('app', 'build.gradle'));
-    if (!appGradle.existsSync()) {
+    if (!appGradleFile.existsSync()) {
```

Comment:
You should rely on getting an error from `readAsLinesSync` instead of checking explicitly first.

The file may exist when this line is executed only to have been deleted by the time the `readAsLinesSync` call is executed.